src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -3,50 +3,112 @@
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List</p>
  */
+
+import java.util.Arrays;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] array = new Object[16];
+    private int indexTopArray = 0;
 
     @Override
     public void add(T value) {
-
+        if (value == null) {
+            return;
+        }
+        if (indexTopArray >= array.length) {
+            array = resizeAndCopyArray(array);
+        }
+        array[indexTopArray] = value;
+        indexTopArray++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (value == null) {
+            return;
+        }
+        if (index < 0 || index > indexTopArray) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        if (indexTopArray >= array.length) {
+            array = resizeAndCopyArray(array);
```

Comment:
can we make method resizeAndCopyArray() void?

Reply:
Yes, I will fix.